Shared/Supporting Files/Views/CategoriesView.swift

```diff
@@ -15,39 +15,74 @@
 import SwiftUI
 
 struct CategoriesView: View {
-    /// The sample categories generated from the samples list.
-    private let categories = Set(SamplesApp.samples.map(\.category)).sorted()
+    /// The visibility of the leading columns in the navigation split view.
+    @Binding var columnVisibility: NavigationSplitViewVisibility
+    
+    /// The selected category loaded from user defaults.
+    @AppStorage("selectedCategory") private var selectedCategory: String?
```

Comment:
That was the intention, yes. Ting and I decided it wouldn't be a good idea to persist a selected sample in case of potential crashes when loading that would then make the app crash upon opening every time. I figured persisting the category was the next best thing, but I can see where you are coming from. Thoughts @yo1995?

Reply:
I'm in favor of not preserving states across multiple app launches. If a user want to have a "shortcut" to a sample, he can favorites it, or don't kill the app and let it stay in the background.